src/fsharp/TypeChecker.fs

```diff
@@ -3099,9 +3099,16 @@ let BuildDisposableCleanup cenv env m (v: Val) =
 let BuildOffsetToStringData cenv env m = 
     let ad = env.eAccessRights
     let offsetToStringDataMethod = 
-        match TryFindIntrinsicOrExtensionMethInfo ResultCollectionSettings.AllResults cenv env m ad "get_OffsetToStringData" cenv.g.system_RuntimeHelpers_ty with 
+        // Prefer string.GetPinnableReference introduced in .NET Core 3
```

Comment:
@cartermp , this is code generation.  So the reflection to find the API, is happening on the machine the compiler is executing on.  So probably we need to generate a helper function to do the call, and then to call that.

Reply:
I don't think I understand. There's no reflection involved here? It's just checking of the type has a given method in its method table that's already been imported.

Reply:
I think you are right, it checks the referenced types rather than the executing types.